src/transaction/log_recovery_redo.hpp

```diff
@@ -644,6 +645,17 @@ void log_rv_redo_record_sync_apply (THREAD_ENTRY *thread_p, log_rv_redo_context
 			 VPID_AS_ARGS (&rcv_vpid), (int) rcv.offset, (int) rcv.length);
     }
 
+  if (is_passive_transaction_server() && log_data.rcvindex == RVPGBUF_DEALLOC)
```

Comment:
I think 
```suggestion
  if (log_data.rcvindex == RVPGBUF_DEALLOC && is_passive_transaction_server())
```
is much more likely to make use of short-circuit evaluation.